### PR TITLE
feat(JobCard): Design adjustments

### PR DESCRIPTION
**Pull request by youwei91**
Check out the requirement listing in this [Trello card](https://trello.com/c/7cObz6pm/870-job-card-ui-enhance)

Update some UI at job card, the changes is apply for DB & JS, so do test on different brands.

I also created a preview environment with this changes, feel free to test it [here](http://srp-preview-update-jobcard-ui.ap-southeast-1.elasticbeanstalk.com)

P.S: in JobCard have many inner component and we should move them into JobCard's component folder instead. I have moved some component related to this PR.

**Old JobCard in Desktop view**
Screen Shot 2019-05-09 at 3 16 59 PM
**New JobCard in Desktop view**
Screen Shot 2019-05-09 at 3 16 52 PM

**Old JobCard in Mobile view**
Screen Shot 2019-05-09 at 3 16 39 PM
**New JobCard in Mobile view**
Screen Shot 2019-05-09 at 3 16 31 PM

### CODE REVIEW

File: react/JobCard/JobCard.less [hunk 1]
```diff
@@ -27,6 +27,12 @@
   flex-direction: column;
 }
 
+.flexRowHeight {
+  @media @desktop {
+    height: 100%;
```
Comment (patrickliu-sa) [inline react/JobCard/JobCard.less]: As the parent div set `display: flex` already, I'll prefer replace `height: 100%` with `flex-grow: 9999` if you want this block take up all the space available vertically

Comment (youwei91) [reply react/JobCard/JobCard.less]: whats the different/benefit of flex-grow:9999 vs height: 100%?

Comment (patrickliu-sa) [reply react/JobCard/JobCard.less]: `Height: 100%` has no relationship with the parent flex container, while using `flex-gain` to state each flex items span ratio is more elegant and recommended

Comment (patrickliu-sa) [reply react/JobCard/JobCard.less]: You can imagine there will be only one `flex-grow: 9999` that take up the most available space while pushing all other sibling flex items away (which have `flex-grow: 1` as default without specified)

**Comment by kosanna** [comment]
I feel like I'm playing a kids' game: find 15 differences between these two pictures 😂

File: react/JobCard/JobCard.js [hunk 2]
```diff
@@ -250,20 +250,22 @@ class JobCard extends React.Component {
             }}
           />
           <Company company={job.company} keyword={keyword} LinkComponent={LinkComponent} trackLinkClicked={trackLinkClicked} />
-          <div className={styles.flexRow}>
+          <div className={classnames(styles.flexRow, styles.flexRowHeight)}>
```
Comment (kosanna) [inline react/JobCard/JobCard.js]: May be better combine those two classes in something more semantically meaningful like `companyMeta`?

File: react/JobCard/JobCard.js [hunk 3]
```diff
@@ -250,20 +250,22 @@ class JobCard extends React.Component {
...
             <div className={styles.leftContent}>
               <MainPoint {...this.props} />
-              <SellingPoint sellingPoints={job.sellingPoints} isSplitView={isSplitView} showSellingPoint={showSellingPoint} enableBrandedAd={enableBrandedAd} />
-              {!isSplitView &&
+              <Hidden mobile>
```
Comment (kosanna) [inline react/JobCard/JobCard.js]: Is that intended to hide selling points and shelf links on mobile for both brands?

File: react/JobCard/JobCard.js [hunk 4]
```diff
@@ -277,14 +279,29 @@ class JobCard extends React.Component {
...
-                />}
+              <ShelfLink
+                job={job}
+                shelfSectionOpen={shelfSectionOpen}
+                desktopOnly
+                onClick={this.handleShelfSectionToggle}
+              />}
             </div>
           </div>
+          <div>
```
Comment (kosanna) [inline react/JobCard/JobCard.js]: Do we need this extra div? Hidden already wraps everything?

File: react/JobCard/JobCard.js [hunk 5]
```diff
@@ -277,14 +279,29 @@ class JobCard extends React.Component {
...
+            <Hidden desktop>
+              <SellingPoint sellingPoints={job.sellingPoints} isSplitView={isSplitView} showSellingPoint={showSellingPoint} enableBrandedAd={enableBrandedAd} />
```
Comment (kosanna) [inline react/JobCard/JobCard.js]: This feels as a common block to above, should we extract it in a component and reuse it?

File: react/JobCard/JobCard.js (hunk 3 above)
Comment (kosanna) [reply react/JobCard/JobCard.js]: Ok, no I see, we are just rearranging them

File: react/JobCard/JobCard.less [hunk 6]
```diff
@@ -41,6 +47,7 @@
 
 .mobileOnly {
   display: flex;
+  padding-top: @grid-base * 2;
```
Comment (kosanna) [inline react/JobCard/JobCard.less]: Are those padding tops semantically linked? If so, we could create a local variable for consistent spacing eg @top-spacing or something, thoughts?

File: react/JobCard/JobCard.js (hunk 2 above)
Comment (youwei91) [reply react/JobCard/JobCard.js]: hmmm 🤔 but for this case we only want here to have the `.flexRowHeight` css, and `.flexRow` is using by others css too for eg: `.container`, so I can't just update the `.flexRow`

File: react/JobCard/JobCard.js (hunk 5 above)
Comment (youwei91) [reply react/JobCard/JobCard.js]: I tried before create another component for SellingPoint, Description, ShelfLink & JobLabel, but I found out there will be a long param to pass in:

```export const Test = ({ ... }) => {
<SellingPoint>
...
}

<Hidden mobile><Test ...></Hidden>
<Hidden desktop><Test ...></Hidden>```

There is total 12 params @@

File: react/JobCard/JobCard.less (hunk 6 above)
Comment (youwei91) [reply react/JobCard/JobCard.less]: not sure about this, but found the padding-top seem using quiet differently is other places

File: react/JobCard/JobCard.js (hunk 2 above)
Comment (kosanna) [reply react/JobCard/JobCard.js]: Ah, I thought it's the only place where .flexRow is used, then it's legitimate

File: react/JobCard/JobCard.js (hunk 5 above)
Comment (kosanna) [reply react/JobCard/JobCard.js]: Yeah that's a pain for all subcomponents. It's not just SellingPoint, it's the whole block that's been added here. We totally need to extract it so it stays in sync with the version above all the time, plus it's making the file more readable and small.

**Pull request merged**